src/utilities/Utilities.cpp

```diff
@@ -134,17 +134,23 @@ namespace Utilities
         /* Get the amount of seconds since the blockchain launched */
 	uint64_t secondsSinceLaunch;
 
-	if (scanHeight < CryptoNote::parameters::DIFFICULTY_TARGET_V2_HEIGHT)
```

Comment:
Same thing applies here as with the days calculation. We can only have a single calculation with `DIFFICULTY_TARGET` if the scan height is < the first change in diff. If our scan height is > 2 or > 3 we need to take into account the two different block speeds.

Probably pretty easy to fix:

```cpp
uint64_t secondsSinceLaunch = 0;

/* How many seconds have passed since DIFFICULTY_TARGET_V3_HEIGHT */
if (scanHeight >= CryptoNote::parameters::DIFFICULTY_TARGET_V3_HEIGHT)
{
    secondsSinceLaunch += (scanHeight - CryptoNote::parameters::DIFFICULTY_TARGET_V3_HEIGHT) * DIFFICULTY_TARGET_V3;
}

/* How many seconds have passed between DIFFICULTY_TARGET_V2_HEIGHT and min(DIFFICULTY_TARGET_V3_HEIGHT, currentHeight) */
if (scanHeight >= CryptoNote::parameters::DIFFICULTY_TARGET_V2_HEIGHT)
{
    secondsSinceLaunch += (std::min(scanHeight, CryptoNote::parameters::DIFFICULTY_TARGET_V3_HEIGHT) - CryptoNote::parameters::DIFFICULTY_TARGET_V2_HEIGHT) * DIFFICULTY_TARGET_V3;
}

/* How many seconds passed between start height and min(scanHeight, DIFFICULTY_TARGET_V2) */
secondsSinceLaunch += std::min(scanHeight, CryptoNote::parameters::DIFFICULTY_TARGET_V2_HEIGHT)  * DIFFICULTY_TARGET_V1;
```

Note: Not tested

Reply:
gotcha, makes sense, we need checking in three blocks now, 1. v3 and above, 2. between v2 and v3 and the original difficulty target up to v2. 

a few finishing touches that build: ``` uint64_t secondsSinceLaunch = 0;

    /* How many seconds have passed since DIFFICULTY_TARGET_V3_HEIGHT */
    if (scanHeight >= CryptoNote::parameters::DIFFICULTY_TARGET_V3_HEIGHT)
    {
        secondsSinceLaunch += (scanHeight - CryptoNote::parameters::DIFFICULTY_TARGET_V3_HEIGHT) * CryptoNote::parameters::DIFFICULTY_TARGET_V3;
    }

    /* How many seconds have passed between DIFFICULTY_TARGET_V2_HEIGHT and min(DIFFICULTY_TARGET_V3_HEIGHT, currentHeight) */
    if (scanHeight >= CryptoNote::parameters::DIFFICULTY_TARGET_V2_HEIGHT)
    {
        secondsSinceLaunch += (std::min(scanHeight, CryptoNote::parameters::DIFFICULTY_TARGET_V3_HEIGHT) - CryptoNote::parameters::DIFFICULTY_TARGET_V2_HEIGHT) * CryptoNote::parameters::DIFFICULTY_TARGET_V3;
    }

    /* How many seconds passed between start height and min(scanHeight, DIFFICULTY_TARGET_V2) */
    secondsSinceLaunch += std::min(scanHeight, CryptoNote::parameters::DIFFICULTY_TARGET_V2_HEIGHT)  * CryptoNote::parameters::DIFFICULTY_TARGET;```

Reply:
Looks good